games/telewave/telewave.py

```diff
@@ -0,0 +1,196 @@
+from chowder import chowder
+
+with open("games/telewave/telewave_config.json", "r") as read_file:
+    config = json.load(read_file)
+
+
+class TelewaveTeam:
+    def __init__(self, players, name, score):
+        self.guessers = deque(players)
+        self.psychic = self.guessers.popleft()
+        self.score = score
+        self.name = name
+
+    def to_string(self):
+        return f"{self.psychic.name}, {', '.join([g.name for g in self.guessers])}"
+
+    def mention_guessers(self):
+        return f"{', '.join([g.mention for g in self.guessers])}"
+
+    def mention(self):
+        return f"{self.psychic.mention}, {', '.join([g.mention for g in self.guessers])}"
+
+    def rotate_psychic(self):
+        self.guessers.append(self.psychic)
+        self.psychic = self.guessers.popleft()
+
+    def get_players(self):
+        return [self.psychic] + list(self.guessers)
+
+
+async def start(bot, ctx, players):
+    await ctx.send(f"Starting a game of **telewave** with {', '.join([p.mention for p in players])}")
+    random.shuffle(players)
+    team_names = get_team_names()
+
+    # Per official Wavelength rules - the team going second starts with a 1 point lead
+    team1 = TelewaveTeam(players[:len(players)//2], name=team_names[0], score=0)
+    team2 = TelewaveTeam(players[len(players)//2:], name=team_names[1], score=1)
+    winners = await play(bot, ctx, team1, team2)
+
+    await ctx.send(f"GG, **{winners.name}** wins")
+    return winners.get_players()
+
+
+async def play(bot, ctx, team1, team2):
+    max_score = config["max_score"]
+    going_again = False
+    while team1.score < max_score and team2.score < max_score:
+        await wait(ctx, going_again, team1)
+        prompt = get_prompt()
+        answer = random.randint(0, 100)
+
+        await display(
+            ctx, team1, team2, prompt, max_score,
+            text=f"\u200B\n\n\n**{team1.psychic.mention}** is thinking of a clue"
+        )
+
+        clue = await get_clue(bot, team1.psychic, prompt, answer)
+
+        # await ctx.send(f"Hold tight {chowder.get_collective_name()}, I just DMed {team1.psychic.mention} the prompt")
+
+        await display(
+            ctx, team1, team2, prompt, max_score,
+            text=f"\u200B\n\n\n**{team1.psychic.mention}**'s clue: \"*{clue}*\""
+        )
+
+        guess = await get_guess(bot, ctx, team1)
+
+        await display(
+            ctx, team1, team2, prompt, max_score,
+            text=f"\u200B\n\n\n**{team1.psychic.mention}**'s clue: \"*{clue}*\"\n"
+                 f"**{team1.name}**'s guess: **{guess}**"
+        )
+
+        counter_guess = await get_counter_guess(bot, ctx, team2)
+
+        prev_score = team1.score
+        result_text = await update_scores(team1, team2, answer, guess, counter_guess)
+        await display(ctx, team1, team2, prompt, max_score, text=result_text)
+
+        team1.rotate_psychic()
+
+        # Per official Wavelength rules - if you score 4 you get to go again
```

Comment:
Which rules?  This one doesn't say anything about this: https://www.ultraboardgames.com/wavelength/game-rules.php

Reply:
It does (see below), but I missed one part - this only applies if you're losing. Will fix.

```
An important catch up rule

No matter how much a team is losing in Wavelength, there's always a chance to catch up. If the Psychic's team scores 4 points, and is still losing to the other team after that round's scoring, they immediately take another turn (with a different player as Psychic).

This means that a team down 9-0 could hypothetically take 3 turns in a row and win.
```